src/main/scala/Main.scala

```diff
@@ -1,8 +1,84 @@
+
+
+import newmodel.Package
+import newmodel._
+
+import scala.collection.mutable.ListBuffer
+
 /**
  * Created by yaroslav on 11.05.2015.
  */
 object Main {
-  def main(args:Array[String]) ={
-    println("hello world")
+
+  object DummyIndex extends Index{
+    val docElements = ListBuffer.empty[DocElement]
+    override def getConversionsFor(id: ID): Seq[ImplicitConversions] = ???
+
+    override def getKnownSubclasses(id: ID): Seq[ID] = ???
+
+    override def getShadowedConversionsFor(id: ID): Seq[ImplicitConversions] = ???
+
+    override def getLinearSupertypesBy(id: ID): Seq[ID] = ???
+
+    override def getCompanionFor(id: ID): ID = ???
+
+    override def getDocElementBy(id: ID): DocElement = docElements.find((x) => x.id == id).get
+
+    def add(element:DocElement): Unit ={
+      docElements += element
+    }
   }
+  def buildModel = {
+
+    val returnForMethod1 = Type("String")
+    val method = MethodDoc("foo", returnForMethod1, None,ID("org.bar.Test#foo"),Comment(""),false,false,false)
+    val constructor = ConstructorDoc("Test", Some(List.empty[ValueParam]),ID("org.bar.Test#<init>"),Comment(""))
+    val classDoc = ClassDoc("Test", List(constructor, method),ID("org.bar.Test"),Comment(""),false)
+    val objectDoc = ObjectDoc("Test",Seq(),ID(""),Comment("org.bar.Test"))
+    val barPackage = Package("bar", List(classDoc),ID("org.bar"),Comment(""))
+    val orgPackage = Package("org", List(barPackage),ID("org"),Comment(""))
+    val rootPackage = Package("_root_", List(orgPackage),ID("_root_"),Comment(""))
```

Comment:
this dummy input should go into a test that calls the generation and compares it to the expected output